src/Components/TodoContainer.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import Header from './Header';
+import InputTodo from './InputTodo';
+import TodosList from './TodosList';
+
+export class TodoContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [],
+    };
+  }
+
+  addTodoItem = (title) => {
+    const newTodoItem = {
+      id: uuidv4(),
+      title,
+      completed: false,
+    };
+    this.setState((state) => ({ todos: [...state.todos, newTodoItem] }));
+  };
+
+  handleChange = (id) => {
+    this.setState((state) => ({
+      todos: state.todos.map((todo) => (todo.id === id ? { ...todo, completed: !todo.completed } : todo)),
+    }));
+  };
+
+  handleDelete = (id) => {
+    this.setState((state) => ({
+      todos: state.todos.filter((todo) => todo.id !== id),
+    }));
+  };
+
+  setUpdate = (updatedTitle, id) => {
+    this.setState({
+      todos: this.state.todos.map((todo) => {
+        if (todo.id === id) {
+          todo.title = updatedTitle;
+        }
+        return todo;
+      }),
+    });
+  };
+
+  componentDidMount() {
+    console.log('component did mount');
+    const temp = localStorage.getItem('todos');
+    const loadedTodos = JSON.parse(temp);
+    if (loadedTodos) {
+      this.setState({
+        todos: loadedTodos,
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    console.log('component Did update');
+    if (prevState.todos !== this.state.todos) {
+      const temp = JSON.stringify(this.state.todos);
+      localStorage.setItem('todos', temp);
+      console.log('function ran');
+    }
+    console.log('component Did update 2');
```

Comment:
Hello @ChaabaneAnas , 
your 'console.log' is supposed to be removed when pushing your work to github to avoid your sensitive information being licked out.

Reply:
Hello @summyalena ,
Thank you for putting in effort to improve my skills .. 
what u said was 100% true ill work on it now